xtask/src/main.rs

```diff
@@ -0,0 +1,599 @@
+                &[],
+            )
+        },
+        "host",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command::<&str>(
+                &[
+                    "cargo",
+                    "test",
+                    "--workspace",
+                    "--features",
+                    "unstable-test",
+                ],
+                None,
+                &[],
+            )
+        },
+        "host",
+        errors,
+    );
+}
+
+fn test_cross(errors: &mut Vec<String>) {
+    println!("🧪 cross");
+    let targets = vec![
+        "thumbv6m-none-eabi",
+        "thumbv8m.base-none-eabi",
+        "riscv32i-unknown-none-elf",
+    ];
+
+    for target in targets {
+        do_test(
+            || {
+                run_command::<&str>(
+                    &["cargo", "check", "--target", &target, "-p", "defmt"],
+                    None,
+                    &[],
+                )
+            },
+            "cross",
+            errors,
+        );
+        do_test(
+            || {
+                run_command::<&str>(
+                    &[
+                        "cargo",
+                        "check",
+                        "--target",
+                        &target,
+                        "-p",
+                        "defmt",
+                        "--features",
+                        "alloc",
+                    ],
+                    None,
+                    &[],
+                )
+            },
+            "cross",
+            errors,
+        );
+    }
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv6m-none-eabi",
+                    "--workspace",
+                    "--exclude",
+                    "defmt-itm",
+                    "--exclude",
+                    "firmware",
+                ],
+                Some("firmware"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv7em-none-eabi",
+                    "--workspace",
+                ],
+                Some("firmware"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command(
+                &[
+                    "cargo",
+                    "check",
+                    "--target",
+                    "thumbv6m-none-eabi",
+                    "--features",
+                    "print-defmt",
+                ],
+                Some("firmware/panic-probe"),
+                &[],
+            )
+        },
+        "cross",
+        errors,
+    );
+
+    do_test(
+        || {
+            run_command::<&str>(
```

Comment:
Not needed since it can be inferred
```suggestion
            run_command(
```